### PR TITLE
[reddit] Add duration to extracted information for posts containing a v.redd.it video

### PR DESCRIPTION
**Pull request by trevnels**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Previously, the Reddit extractor would not provide video duration for post videos served through v.redd.it, though this information is provided by the Reddit post API. This change includes the duration key of the API response in the video's extracted information when present. I also added duration to the existing test's info_dict for coverage.

### CODE REVIEW

**Comment by dstftw** [comment]
Read coding conventions.

**Comment by trevnels** [comment]
@dstftw I believe this is now correct.

File: youtube_dl/extractor/reddit.py [hunk 1]
```diff
@@ -116,6 +117,10 @@ def _real_extract(self, url):
         else:
             age_limit = None
 
+        duration = None
+        if data.get('media') and data['media'].get('reddit_video'):
+            duration = data['media']['reddit_video'].get('duration')
```
Comment (dstftw) [inline youtube_dl/extractor/reddit.py:122]: Breaks.

Comment (trevnels) [reply youtube_dl/extractor/reddit.py]: Replaced with try_get

**Comment by dstftw** [comment]
>- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

**Pull request merged**